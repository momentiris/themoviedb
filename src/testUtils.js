import * as React from 'react'
import { Context, initialState } from './components/contextProvider'
import { BrowserRouter as Router } from 'react-router-dom'

import { render as rtlRender } from 'react-testing-library'

export const dispatch = jest.fn()
export const state = initialState

export const render = (children, state = initialState) => {
  return rtlRender(
    <Context.Provider value={{ dispatch, state }}>
      <Router>{children}</Router>
    </Context.Provider>
  )
}
