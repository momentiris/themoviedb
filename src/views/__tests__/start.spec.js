import React from 'react'
import { render, dispatch, initialState } from '../../testUtils'
import Start from '../../views/start'
import { fireEvent, cleanup } from 'react-testing-library'

describe('views/start', () => {
  afterEach(cleanup) // <-- add this
  it('renders component', () => {
    const { container } = render(<Start />, initialState)
    expect(container).toMatchSnapshot()
  })

  it('rerenders with movies given updated query', () => {
    const mockedMoviesState = {
      ...initialState,
      searchResult: {
        movies: [
          {
            backdrop_path: '123',
            title: 'Lasse Kongo Returns',
          },
          {
            backdrop_path: '123',
            title: 'Lasse Kongo Returns',
          },
        ],
      },
    }
    const { container } = render(<Start />, initialState)
    expect(container).toMatchSnapshot()

    const searchBar = container.querySelector('input')
    fireEvent.change(searchBar, { target: { value: 'Batman' } })
    fireEvent.click(container.querySelector('button'))
    expect(dispatch).toHaveBeenCalled()

    const { container: updatedContainer } = render(<Start />, mockedMoviesState)
    expect(updatedContainer).toMatchSnapshot()
  })
})
