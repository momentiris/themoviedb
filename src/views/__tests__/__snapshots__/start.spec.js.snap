// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/start renders component 1`] = `
<div>
  <div
    class="sc-EHOje bsGAgl"
  >
    <header
      class="sc-htoDjs gxcVEN"
    >
      Find a companion to your lonely evening.
    </header>
    <div
      class="sc-bdVaJa cdlyjo"
    >
      <input
        class="sc-bwzfXH fodZgz"
        placeholder="Search for something... or don't, I don't care."
      />
      <button
        class="sc-htpNat fQXDrD"
      >
        Search
      </button>
    </div>
    
    <ul
      class="sc-ifAKCX hBLwFX"
    />
  </div>
</div>
`;

exports[`views/start rerenders with movies given updated query 1`] = `
<div>
  <div
    class="sc-EHOje bsGAgl"
  >
    <header
      class="sc-htoDjs gxcVEN"
    >
      Find a companion to your lonely evening.
    </header>
    <div
      class="sc-bdVaJa cdlyjo"
    >
      <input
        class="sc-bwzfXH fodZgz"
        placeholder="Search for something... or don't, I don't care."
      />
      <button
        class="sc-htpNat fQXDrD"
      >
        Search
      </button>
    </div>
    
    <ul
      class="sc-ifAKCX hBLwFX"
    />
  </div>
</div>
`;

exports[`views/start rerenders with movies given updated query 2`] = `
<div>
  <div
    class="sc-EHOje bsGAgl"
  >
    <header
      class="sc-htoDjs gxcVEN"
    >
      Find a companion to your lonely evening.
    </header>
    <div
      class="sc-bdVaJa cdlyjo"
    >
      <input
        class="sc-bwzfXH fodZgz"
        placeholder="Search for something... or don't, I don't care."
      />
      <button
        class="sc-htpNat fQXDrD"
      >
        Search
      </button>
    </div>
    <ul
      class="sc-ifAKCX hBLwFX"
    >
      <a
        href="/movie/undefined"
        style="text-decoration: none; color: black;"
      >
        <li
          class="sc-bxivhb jekTJH"
          title="Lasse Kongo Returns"
        >
          <span>
            Lasse Kongo Returns
          </span>
          <span>
            Released: 
          </span>
          <span>
            Score: 
          </span>
        </li>
      </a>
      <a
        href="/movie/undefined"
        style="text-decoration: none; color: black;"
      >
        <li
          class="sc-bxivhb jekTJH"
          title="Lasse Kongo Returns"
        >
          <span>
            Lasse Kongo Returns
          </span>
          <span>
            Released: 
          </span>
          <span>
            Score: 
          </span>
        </li>
      </a>
    </ul>
  </div>
</div>
`;
